lib/sycamore/sycamore/transforms/sort.py

```diff
@@ -0,0 +1,57 @@
+from typing import Any, Optional
+
+from sycamore.plan_nodes import Node, Transform
+from sycamore.data import Document
+
+from ray.data import Dataset
+
+
+class Sort(Transform):
+    """
+    Sort by field in Document
+    """
+
+    def __init__(self, child: Node, descending: bool, field: str, default_val: Optional[Any] = None):
+        super().__init__(child)
+        self._descending = descending
+        self._field = field
+        self._default_val = default_val
+
+    def execute(self, **kwargs) -> "Dataset":
+        # creates dataset
+        ds = self.child().execute(**kwargs)
+
+        # adds a "key" column containing desired field
+        map_fn = self.make_map_fn_sort()
+        ds = ds.map(map_fn)
```

Comment:
Are metadata documents included here? If so, how do they get sorted?

Reply:
Actually have not tested with metadata documents — how can this be done? I'm thinking that having a default value should deal with them though